pkg/route/validation/validation_test.go

```diff
@@ -65,6 +64,9 @@ func createRouteSpecTo(name string, kind string) routev1.RouteTargetReference {
 // TestValidateRoute ensures not specifying a required field results in error and a fully specified
 // route passes successfully
 func TestValidateRoute(t *testing.T) {
```

Comment:
You should instead write a test for `validateHeaders` and run through many more scenarios, and check specifics in the error messages.

Reply:
done